crypto/context.c

```diff
@@ -232,7 +232,12 @@ static void ossl_lib_ctx_generic_new(void *parent_ign, void *ptr_ign,
     void *ptr = meth->new_func(ctx);
 
     if (ptr != NULL) {
-        CRYPTO_THREAD_write_lock(ctx->lock);
+        if (!CRYPTO_THREAD_write_lock(ctx->lock))
+            /*
+             * Can't return something, so best to hope that something will
+             * fail later. :(
+             */
```

Comment:
Is there a reason you can't make this function return something?

Reply:
it's a `new_func` function used in the crypto "exdata" stuff.  Look how "new_func" is used in `crypto/ex_data.c`  Changing it to non-void would be lots of work and subtly breaking things.